modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/transaction/TransactionStorage.scala

```diff
@@ -142,6 +143,11 @@ class TransactionStorage[F[_]: Async: KryoSerializer](
       }.map(_.flatten)
     } yield NonEmptyList.fromList(allPulled)
 
+  def find(hash: Hash): F[Option[Hashed[Transaction]]] =
+    waitingTransactions.toMap.flatMap { _.view.values.toList.flatMap(_.toList).traverse(_.toHashed) }.map {
```

Comment:
Looks like a very costly operation. If we need to allow users to fetch the waiting transaction by hash, then maybe we should store it differently (with hash next to it).

Reply:
Maybe as `Hashed[Transaction]` even

Reply:
We are generating the hash either way when uploading the transaction, so we can as well store it is `Hashed`, then the consecutive function and this functionality should get simpler.

Reply:
I think it may be fine to use `Hashed` internally, sending already `Hashed` value to someone would be problematic (everyone should verify Signed on their own)